compiler/rustc_mir/src/interpret/intrinsics.rs

```diff
@@ -435,6 +435,12 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 // These just return their argument
                 self.copy_op(args[0], dest)?;
             }
+            sym::assume => {
+                let cond = self.read_scalar(args[0])?.check_init()?.to_bool()?;
+                if !cond {
+                    throw_ub_format!("`assume` intrinsic called with `false`");
+                }
+            }
```

Comment:
Please remember to also remove [the Miri implementation of this](https://github.com/rust-lang/miri/blob/1f6ce43721bb13afeda2b91dea2db80b89cea37b/src/shims/intrinsics.rs#L481) once the PR lands.

Reply:
yeah, I'm preparing PR for that.